Washington-County/COUNTY-DRUPAL
# WCOR-20405: Configure a starter structure for group page menu views and breadcrumbs.

### Use Case

DigEx Team can see the basic page, resource, and group menu views and path alias settings that allow first draft of content to be added to County Drupal in a hierarchical structure.

**Implementation Details**

* Added field_states_ui and subpathauto to allow for menu UI and breadcrumb creation from new paths.
* Enabled help module to make it easier to find tokens (and solved the mystery of demonstrating block regions in the theme :o)
* Ordered fields for forms and default/teaser displays for the following:
** Groups
** Pages
** Resources
* Refactored resource description to use existing field_summary (missed that in the spec originally)
* Added help text for menu-related fields and turned on visibility controls for "show in menu" targeting sort weight (new field) and menu parent.
* Added entity reference view to restrict menu parent to pages or resources that do not have a parent already—and added path to help editor selection of correct content.
* Configured pathauto patterns to spec for the following:
** Alerts
** Contacts
** Groups
** Media (single pattern to rule them all)
** News
** Notifications
** Pages
** Resources
** Terms (default pattern for many terms)
** Users
* Set pathauto patterns to allow field_group_path as a safe token that can contain slashes.
* Added group_page_menu view that adds block menu views for group homepages, parent page showing children, and siblings of a common parent page. This view also has permission restricted management buttons that link to an admin view.

### Test Plan

`acli remote:drush @wcor.ode56 uli`

1. Add at least 2 pages to an existing group with "Show in menu" checked and no parent defined. If desired, make the sort weight difference override the short title alphabetical sort.
1. Add at least 2 more pages to the group with "Show in menu" checked and with one of the previous pages set as the menu parent.
1. Check that when viewing these pages a menu appears that allows you to see all top level pages from the group homepage, as well as the child pages of the parent page, and the siblings of a child with a link back to the parent.
1. Ensure that the deepest level page has breadcrumbs and a pathauto alias that includes its parent page.
1. Create at least one item of each of the types listed in pathauto patterns in the above implementation details. Ensure the paths make sense per the spec.
